### PR TITLE
Length units

### PR DESCRIPTION
**Pull request by jessfraz**
to catch you up @adamchalmers (and im happy to go over on a call) this PR https://github.com/KittyCAD/engine/pull/1780 means the engine assumes all units are mm. The easiest way for us to then use kcl with units, in the current setup without having to change both the client side js to be unit aware, and the executor to be unit aware, is to change the rust actor to be unit aware and convert right before sending to the engine to mm.

So assume we have a modeling command we fire right at start of engine: 
```SetUnits{units: UnitLength}``` we can also call this when the user changes their settings. regardless it changes the units in the engine rust context, then from there we convert in the actor for anything that uses `LengthUnit` the wrapper around an f64. 

Open to other names I don't really care but yeah this is what we landed on last week as for the implementation

TODO:
- [x] add the new type f64 wrapper
- [x] make sure we change every f64 that needs to be changed to the new type
- [x] add `SetUnits` modeling cmd
- [x] pull thru to engine and make sure it all works

**Note to self:** if the user changes the units in their settings mid scene we will need to re-execute from modeling app.

### CODE REVIEW

**Comment by jessfraz** [comment]
added all the reviewers to make sure I didn't miss an endpoint that needs updating or that i didn't update something that is not actually a unit

File: modeling-cmds/src/length_unit.rs [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+}
+
+impl Point4d<LengthUnit> {
+    /// Convert the point to millimeters.
+    pub fn to_millimeters(&self, from: crate::units::UnitLength) -> Point4d<f64> {
+        Point4d {
+            x: self.x.to_millimeters(from),
+            y: self.y.to_millimeters(from),
+            z: self.z.to_millimeters(from),
+            w: self.w.to_millimeters(from),
```
Comment (alteous) [inline modeling-cmds/src/length_unit.rs:74]: The `w` component should be left as-is.

Comment (jessfraz) [reply modeling-cmds/src/length_unit.rs:74]: good find

File: modeling-cmds/src/each_cmd.rs [hunk 2]
```diff
@@ -720,6 +723,14 @@ pub struct ImportFile {
     pub data: Vec<u8>,
 }
 
+/// Set the units of the scene.
+/// For all following commands, the units will be interpreted as the given units.
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
+pub struct SetSceneUnits {
+    /// The unit to interpret the scene as.
```
Comment (adamchalmers) [inline modeling-cmds/src/each_cmd.rs:730]:
```suggestion
    /// Which units the scene uses.
```

**Review by adamchalmers — CHANGES_REQUESTED** [review]

File: Cargo.lock
[the remark this answers is not in the document]
Comment (jessfraz) [reply Cargo.lock:1]: was unintentional

**Review by adamchalmers — APPROVED** [review]

**Pull request merged**